pkg/agent/handlers/execute.go

```diff
@@ -0,0 +1,52 @@
+package handlers
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	testworkflowsv1 "github.com/kubeshop/testkube-operator/pkg/client/testworkflows/v1"
+	"github.com/kubeshop/testkube/pkg/agent"
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	"github.com/kubeshop/testkube/pkg/cloud"
+	"github.com/kubeshop/testkube/pkg/testworkflows/testworkflowexecutor"
+	"github.com/kubeshop/testkube/pkg/utils/codec"
+)
+
+// TODO - valid error handling
+
+func NewExecuteTestWorkflowHandler(
```

Comment:
This is for sure not needed:
* There is already option to execute Test Workflow via generic gRPC command we have (that calls API)
* For future Execution Worker, the command will look differently - as the Execution Worker needs to be stateless

Reply:
I've planned here to fully decouple from API server - as it's additional unneded thing.

Reply:
These was the attempt to decouple it from the API server totally - can be refactored/reordered if we decide about API later

Reply:
> I've planned here to fully decouple from API server - as it's additional unneded thing. So it's quite needed to not spawn this APIServer

But when it will be decoupled from the API Server, the signature will also differ, so (A) this function is for decoupling, yet (B) it needs to be deleted after decoupling, as it will look differently 🙂